src/utils/factorization/from.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementations for a [`Factorization`] of [`Z`] values.
+
+use super::Factorization;
+use crate::{integer::Z, traits::AsInteger};
+use flint_sys::fmpz_factor::{_fmpz_factor_append, fmpz_factor_get_fmpz};
+
+impl<Integer: AsInteger + Into<Z>> From<Integer> for Factorization {
+    /// Convert an integer into a [`Factorization`].
+    ///
+    /// Parameters:
+    /// - `factor`: the integer we want as a [`Factorization`] object.
+    ///
+    /// Returns a new [`Factorization`] with `factor` as the only factor.
+    ///
+    /// # Examples:
+    /// ```
+    /// use qfall_math::utils::Factorization;
+    ///
+    /// let fac = Factorization::from(10);
+    /// ```
+    fn from(factor: Integer) -> Self {
+        let factor = factor.into();
+        let mut out = Self::default();
+        unsafe { _fmpz_factor_append(&mut out.factors, &factor.value, 1) };
+        out
+    }
+}
+
+impl<Integer: AsInteger + Into<Z>> From<(Integer, Integer)> for Factorization {
+    /// Convert two integers into a [`Factorization`].
+    ///
+    /// Note that the order of the factors is not altered by this method.
+    ///
+    /// Parameters:
+    /// - `factors`: a tuple with two integers we want in the [`Factorization`] object.
+    ///
+    /// Returns a new [`Factorization`] with `factor` as the only factor.
+    ///
+    /// # Examples:
+    /// ```
+    /// use qfall_math::utils::Factorization;
+    ///
+    /// let fac = Factorization::from((10, 5));
+    /// ```
+    fn from(factors: (Integer, Integer)) -> Self {
+        let factor_1 = factors.0.into();
+        let factor_2 = factors.1.into();
+        let mut out = Self::default();
+        unsafe { _fmpz_factor_append(&mut out.factors, &factor_1.value, 1) };
+        unsafe { _fmpz_factor_append(&mut out.factors, &factor_2.value, 1) };
+        out
+    }
+}
+
+impl From<&Factorization> for Vec<(Z, u64)> {
```

Comment:
if this is implemented in one direction, I think it should be implemented in the other as well, i.e. From<&Vec<(Z, u64)>> for Factorization

Reply:
done